00XX-common-ledger-protocol/00XX-common-ledger-protocol.md

```diff
@@ -0,0 +1,273 @@
+# Common Ledger Protocol (CLP)
+
+## Preface
+
+This document describes the Common Ledger Protocol (CLP), a ledger protocol for
+bilateral liquidity relationships. It is similar in function to [Plugin RPC],
+but has been rewritten to use OER instead of JSON, and extensibility features
+have been added.
+
+## Introduction
+
+### Motivation
+
+There are many different types of ledgers, and Interledger aims to interoperate
+all of them. For fast ledgers that support hashlocks, it is enough to wrap the
+ledger's API in a [Ledger Plugin]. These allow you to send interledger payments
+through anyone else on that ledger who is running your plugin.
+
+In lots of scenarios, we don't have an underlying ledger that's fast enough to
+do every ILP payment on-ledger. There are also many situations when two parties
+trust one another up to an amount, and don't need any third-party ledger to
+book-keep for them. Currently, this need is filled by [ILP Plugin Virtual],
+which uses [Plugin RPC] as its ledger protocol. JSON messages are passed back
+and forth to perform transfers and send quote requests. Custom JSON messages
+can be added to extend the Plugin RPC messages, but they aren't well defined.
+
+Because the use case for a bilateral ledger protocols is so ubiquitous, the CLP
+has been invented to make the protocol extensible, efficient, and friendly to
+re-implementation.
+
+### Scope
+
+The CLP, despite its name, is just another ledger protocol. It manages
+conditional transfers, messaging requests, and allows [side protocols] for
+extensibility. You can use ILP without using CLP. CLP is intended to be a
+well-suited solution so that a new bilateral ledger protocol doesn't need to
+exist for every new use case.
+
+This document describes the flow and messages that CLP uses. It does not cover
+establishment of CLP channels, nor does it cover any side protocols.  Side
+protocols include functionality like ledger metadata, balance, automated
+settlement, and dispute resolution. 
+
+The CLP Packet format is described exactly in the [CLP ASN.1 spec].
+
+## Terminology
+
+- **CLP** is the common ledger protocol, as described by this packet and the
+  ASN.1 spec.
+
+- A **Side Protocol** is a protocol which isn't defined by CLP and is carried
+  in the side protocol data (see below).
+
+- A **CLP Connection** is a websocket connection over which CLP Packets are
+  sent.
+
+- **CLP Packets** are the messages defined in the CLP ASN.1 spec.
+
+- **Peers** are the parties on a CLP connection. Your peer is the party on the
```

Comment:
Are you using the terms **CLP connection** and **CLP channel** interchangeably?

Earlier you talk about establishing a CLP channel, which I interpret as something logical (i.e. the physical connection could be anything, could go through proxies etc.)

I think of peers as logical entities on either side of a channel as opposed to some systems on either side of the connection. Which do you mean?

Reply:
Peer is confusing from my point of view. They are just peers at the websocket transport layer, but one is acting as server and the other one is a client, even if a given setup allows both CLP hosts to act as client and server for non-related messages.